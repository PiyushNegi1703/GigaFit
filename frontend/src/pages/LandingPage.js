import { Link } from "react-router-dom";
import desktopImage from "../assets/login 1.png";
// import mobileImage from "../assets/login mobile.png";
import logo from "../assets/logo.png";

const LandingPage = () => {
  return (
    <div className="landing-page" data-testid="land">
        <div className="image-container">
          <img
            src={desktopImage}
            alt="bg img"
            className="background-image"
          />

          <div
            className="logo-container"
          >
            <img
              src={logo}
              alt="logo"
            />
          </div>

          <div
            className="text-container"
          >
            <p data-testid="text">
              Transform your Body, Mind and Soul with <br /> GIGAFIT
            </p>
          </div>

          <div
            className="button-container"
          >
            <Link to="/login">
              <button className="button" data-testid="button-text">Get Started</button>
            </Link>
          </div>
        </div>
    </div>
  );
};

export default LandingPage;
