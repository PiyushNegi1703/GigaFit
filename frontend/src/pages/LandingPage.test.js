import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import LandingPage from "./LandingPage";
import {BrowserRouter as Router} from 'react-router-dom';
import { isCompositeComponent, isDOMComponent, isElement, isElementOfType } from "react-dom/test-utils";

afterEach(() => {
    cleanup();
})

test('should render LandingPage component', () => {
  render(
    <Router>
        <LandingPage />
    </Router>
  );
  const landingElement = screen.getByTestId('land');
  expect(landingElement).toBeInTheDocument();
})

test('should have the same text component', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );
      const textElement = screen.getByTestId('text');
      expect(textElement).toHaveTextContent('Transform your Body, Mind and Soul with GIGAFIT')
})

test('should have the button text', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );

      const buttonElement = screen.getByTestId('button-text')
      expect(buttonElement).toHaveTextContent('Get Started')
})

test('should match snapshot', () => {
  const tree = renderer.create(<Router>
    <LandingPage />
</Router>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('should not be integer', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );

      const buttonElement = screen.getByTestId('button-text')
      expect(isNaN(buttonElement)).toBe(true)
})

test('should be non-user defined element', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );

      const buttonElement = screen.getByTestId('button-text')
      expect(isCompositeComponent(buttonElement)).toBe(false)
})

test('should be a DOM element', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );

      const buttonElement = screen.getByTestId('button-text')
      expect(isDOMComponent(buttonElement)).toBe(true)
})

test('should not be a React element', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );

      const buttonElement = screen.getByTestId('button-text')
      expect(isElement(buttonElement)).toBe(false)
})

test('should not be a react component', () => {
    render(
      <Router>
          <LandingPage />
      </Router>
    );
    const landingElement = screen.getByTestId('land');
    expect(isElement(landingElement)).toBe(false);
  })

test('should not be of type componentClass', () => {
    render(
        <Router>
            <LandingPage />
        </Router>
      );
      const landingElement = screen.getByTestId('land');
    expect(isElementOfType(landingElement)).toBe(false);
})