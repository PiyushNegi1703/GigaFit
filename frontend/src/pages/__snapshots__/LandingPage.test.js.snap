// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="landing-page"
  data-testid="land"
>
  <div
    className="image-container"
  >
    <img
      alt="bg img"
      className="background-image"
      src="login 1.png"
    />
    <div
      className="logo-container"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </div>
    <div
      className="text-container"
    >
      <p
        data-testid="text"
      >
        Transform your Body, Mind and Soul with 
        <br />
         GIGAFIT
      </p>
    </div>
    <div
      className="button-container"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="button"
          data-testid="button-text"
        >
          Get Started
        </button>
      </a>
    </div>
  </div>
</div>
`;
